src/modules/auth/auth.service.ts

```diff
@@ -212,4 +214,96 @@ export class AuthService {
       };
     }
   }
+
+  async forgotPassword(email: string): Promise<Result> {
+    try {
+      const user = await this.userService.findByEmail(email);
+      if (!user) {
+        return {
+          code: RESET_PASSWORD_ERROR,
+          message: "account doesn't exist",
+        };
+      }
+
+      const resetPasswordToken = this.emailVerificationService.generateVerificationToken(email);
+      const res = await this.userService.update(user.id, {
+        resetPasswordToken,
+      });
+      const requestResetUser = await this.userService.findByEmail(email);
+      if (res) {
+        await this.emailVerificationService.sendResetPasswordEmail(requestResetUser);
+        return {
+          code: SUCCESS,
+          message: 'We have sent you a reset email. Please check your mailbox.',
+        };
+      }
+    } catch (error) {
+      return {
+        code: RESET_PASSWORD_ERROR,
+        message: 'An unexpected error occurred during sending reset password email.',
+      };
+    }
+  }
+
+  async resetPassword(input: ResetPasswordInput): Promise<Result> {
+    try {
+      const { newPassword, resetToken } = input;
```

Comment:
resetToken should not be inside the input, it's used to be gotten from header.

And validate token should be a step prior to resetPassword process.